### ISSUE TITLE
Do not manage to use coverage and line_profiler

### ISSUE THREAD

**Issue by vpicouet** [coveragepy/coveragepy#1183]
Hello,
I am trying to implement a test code that would also give the time profile when ran with coverage.
I realized that code executed with exec are excluded from coverage (https://github.com/nedbat/coveragepy/issues/380)
but bypassing exec, the executed lines are still not flagged as covered.
Here is an example script with line_profile https://github.com/pyutils/line_profiler:
```
from line_profiler import LineProfiler
import pstats
from io import StringIO

def printer(value):
    print(value)
    return

file='/tmp/profile.py'

def Profile_new(command,functions=[],args=None):
    result=StringIO()

    line_prof = LineProfiler(*functions)
    a = line_prof.runcall(command,*args)#
    line_prof.print_stats(result)    
    result=result.getvalue()
    print(result)
    print('This is compiled but do not appear in coverage...')
    f=open(file,'w')
    f.write(result)
    f.close()
    return line_prof

Profile_new(printer,args=('hello world',),functions=[printer,])
```

I run this python script with: `coverage run my_script.py` & `coverage html -d /tmp/profile/`
Weirdly the print function in printer is not flagged as covered, as I would expect it to be. The result is the same if printer function is in another file and importer in my_script.py.

![Capture d’écran, le 2021-06-29 à 01 20 58](https://user-images.githubusercontent.com/37241971/123715297-44fc6980-d878-11eb-819f-dadb90cff78d.jpg)

**To Reproduce**
python version:  3.8
coverage version:  5.3

I am still a beginner with coverage so I must be wrong, but is there then a way to combine coverage and profiling in an efficient way? I could not find any good or simple solution.
Thanks a lot for your help!

**Comment by nedbat** [comment]
Profiling and coverage use very similar underlying mechanisms to do their measurements, so it doesn't surprise me that they interfere with each other.  I wouldn't think profiling during coverage would be useful anyway: coverage adds overhead to the execution, which will change the timing numbers.  Can you say more about why you want to do them together?

**Comment by vpicouet** [comment]
Ok, I understand.
The goal would be develop progressively a python test file to test a python package and get both profiling and coverage.
I feel that it is simpler and more direct to have only one set of test that can provide both information instead of developing two series of tests for the 2 tasks.
I understand that coverage might add some extra time to profiling but I don't think it would be an issue for me as I only need profiling to spot the most time consuming lines (in order to try finding alternative way of coding it if needed) not to get the absolute timing of each line.
Do you think it has already been done?

**Comment by nedbat** [comment]
You can still use one test suite to measure both coverage and profiling, but you have to run it twice, once for each kind of measurement.  Would that work for you?

**Comment by vpicouet** [comment]
That would be perfect! 
But I actually did not manage to do that with [line_profiler](https://github.com/rkern/line_profiler), which I like because you only profile selected functions.
There are 3 ways to use it:
-with `%lprun` but this work only with python and interfere with coverage
-via `kernprof -l script_to_profile.py`  but that requires to add `@profile` on all the functions of that package you want to profile which is kind of invasive...
-importing LineProfiler which I tried (See my first comment) but then coverage does not work.
If you have an example of profiler and a command line I could directly run on a regular test suite, that would be great.
Thanks again for your help!
